.github/workflows/golangci-lint.yml

```diff
@@ -17,10 +17,8 @@ jobs:
     steps:
       - uses: actions/checkout@v4
       - uses: actions/setup-go@v5
-        with:
-          go-version-file: 'go.mod'
       - name: golangci-lint
-        uses: golangci/golangci-lint-action@3cfe3a4abbb849e10058ce4af15d205b6da42804 # v4
+        uses: golangci/golangci-lint-action@v6
```

Comment:
Can we use the pinned commit for v6? 🙏 It's recommended for not finding surprises in undesired bumps

Reply:
I can but I tend to take opposite view, trust developers to keep compatibility in major version allowing you to get backwards compatible fixes and enhancements as they are released, reducing the maintenance overhead and increasing readability. 

This is also inline with the other use directives above.

Thoughts?

Reply:
It's a practice I try to follow regarding security: https://www.stepsecurity.io/blog/pinning-github-actions-for-enhanced-security-a-complete-guide

Reply:
> It's a practice I try to follow regarding security: https://www.stepsecurity.io/blog/pinning-github-actions-for-enhanced-security-a-complete-guide

I suspected that might be the case, updated.